Tutorial.playground/Contents.swift

```diff
@@ -0,0 +1,484 @@
+let characterArray = uppersAndLowers.proliferate()
+
+characterArray.generate
+characterArray.generate
+characterArray.generate
+
+/// `proliferateNonEmpty` works like `proliferate` but guarantees the generated array is never empty.
+let oddLengthArrays = fromOnetoFive.proliferateNonEmpty().suchThat { $0.count % 2 == 1 }
+
+oddLengthArrays.generate.count
+oddLengthArrays.generate.count
+oddLengthArrays.generate.count
+
+//: Generators also admit functional methods like `map` and `flatMap`, but with different names than you might
+//: be used to.
+
+// `fmap` (function map) works exactly like Array's `map` method; it applies the function to any 
+// values it generates.
+let fromTwoToSix = fromOnetoFive.fmap { $0 + 1 }
+
+fromTwoToSix.generate
+fromTwoToSix.generate
+fromTwoToSix.generate
+fromTwoToSix.generate
+fromTwoToSix.generate
+
+// `bind` works exactly like Array's `flatMap`, but instead of concatenating the generated arrays it produces
+// a new generator that picks values from among the newly created generators produced by the function.  While
+// That definition may *technically* be what occurs, it is better to think of `bind` as a way of making a generator
+// depend on another.  For example, you can use a generator of sizes to limit the length of generators of arrays:
+
+let generatorBoundedSizeArrays = fromOnetoFive.bind { len in
+	return characterArray.suchThat { xs in xs.count <= len }
+}
+
+generatorBoundedSizeArrays.generate
+generatorBoundedSizeArrays.generate
+generatorBoundedSizeArrays.generate
+generatorBoundedSizeArrays.generate
+generatorBoundedSizeArrays.generate
+
+//: Because SwiftCheck is based on the functional concepts in our other library [Swiftz](https://github.com/typelift/Swiftz),
+//: each of these functions has an operator alias:
+//:
+//: * `<^>` is an alias for `fmap`
+//: * `<*>` is an alias for `ap`
+//: * `>>-` is an alias for `bind`
+
+// <^> is backwards for aesthetic and historical purposes.  Its true use will be revealled soon.
+let fromTwoToSix_ = { $0 + 1 } <^> fromOnetoFive
+
+fromTwoToSix_.generate
+fromTwoToSix_.generate
+fromTwoToSix_.generate
+
+let generatorBoundedSizeArrays_ = fromOnetoFive >>- { len in
+	return characterArray.suchThat { xs in xs.count <= len }
+}
+
+generatorBoundedSizeArrays_.generate
+generatorBoundedSizeArrays_.generate
+generatorBoundedSizeArrays_.generate
+
+//: Now that you've seen what generators can do, we'll use all we've learned to create a generator that
+//: produces email addresses.  To do this, we'll need one more operator/method notated `<*>` or `ap`.
+//: `ap` comes from [Applicative Functors](http://staff.city.ac.uk/~ross/papers/Applicative.html) and is
+//: used to "zip together" `Gen`erators of functions with `Gen`erators of of values, applying each function
+//: during the zipping phase.  That definition is a little hand-wavey and technical, so for now we'll say that
+//: `ap` works like "glue" that sticks special kinds of generators togethers.
```

Comment:
This is a dumb and lazy and inaccurate simplification.  I think I'll use a diagram.